flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/sources/TableSourceUtil.scala

```diff
@@ -100,18 +96,32 @@ object TableSourceUtil {
           throw new ValidationException(s"Rowtime field '$name' has invalid type $t. " +
             s"Rowtime attributes must be of TimestampType.")
         }
-        val (physicalName, idx, tpe) = resolveInputField(name, tableSource)
+        val (physicalName, idx, logicalType) = resolveInputField(name, tableSource)
         // validate that mapped fields are are same type
-        if (!isAssignable(fromTypeInfoToLogicalType(tpe), t)) {
+        if (!isAssignable(logicalType, t)) {
           throw new ValidationException(s"Type $t of table field '$name' does not " +
-            s"match with type $tpe of the field '$physicalName' of the TableSource return type.")
+            s"match with type $logicalType of the field '$physicalName' of the " +
+            "TableSource return type.")
+        } else if (!isInteroperable(logicalType, t)) {
```

Comment:
Why check decimal precision?

Reply:
As we discussed offline, if the produced data type of TableSource doesn't carry the correct precision, a readable exception should be thrown. This is a guide to suggest users to update their connectors.